pcraster_tools/processing/algorithms/map2col.py

```diff
@@ -0,0 +1,89 @@
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************
+"""
+
+import os
+
+from qgis.core import (QgsProcessing,
+                       QgsProcessingParameterMultipleLayers,
+                       QgsProcessingParameterFileDestination,
+                       )
+
+from pcraster_tools.processing.algorithm import PCRasterAlgorithm
+
+
+class Map2colAlgorithm(PCRasterAlgorithm):
+    """
+    Converts from PCRaster map format to column file format.
+    """
+
+    INPUT_RASTERS = 'INPUT'
+    OUTPUT_CSV = 'OUTPUT'
+
+    def createInstance(self):  # pylint: disable=missing-function-docstring
+        return Map2colAlgorithm()
+
+    def name(self):  # pylint: disable=missing-function-docstring
+        return 'map2col'
+
+    def displayName(self):  # pylint: disable=missing-function-docstring
+        return self.tr('PCRaster maps to column file')
+
+    def group(self):  # pylint: disable=missing-function-docstring
+        return self.tr('Data management')
+
+    def groupId(self):  # pylint: disable=missing-function-docstring
+        return self.tr('data')
+
+    def shortHelpString(self):  # pylint: disable=missing-function-docstring
+        return self.tr(
+            """Converts from PCRaster map format to column file format (csv).
+
+            <a href="{}">PCRaster documentation</a>
+
+            Parameters:
+
+            * <b>Input Raster layers</b> (required) - raster layers from any data type. The maps must have the same projection, the other location attributes (use the resample tool if this is not the case) and the data types may be different between the maps.
+            * <b>Output text file</b> (required) - text file with comma separated columns
+            """
+        ).format(PCRasterAlgorithm.documentation_url('app_map2col.html'))
+
+    def initAlgorithm(self, config=None):  # pylint: disable=missing-function-docstring,unused-argument
+        self.addParameter(
+            QgsProcessingParameterMultipleLayers(
+                self.INPUT_RASTERS,
+                self.tr('Input Raster Layer(s)'),
+                QgsProcessing.TypeRaster
+            )
+        )
+
+        self.addParameter(
+                QgsProcessingParameterFileDestination(
+                self.OUTPUT_CSV,
+                self.tr('Output textfile with columns'),
+                defaultValue=None,
+                optional=False
+            )
+        )
+
+    def processAlgorithm(self, parameters, context, feedback):  # pylint: disable=missing-function-docstring,unused-argument,too-many-locals
+        input_rasters = []
+        for layer in self.parameterAsLayerList(parameters, self.INPUT_RASTERS, context):
```

Comment:
There's nothing here stopping a user from selecting a non-map raster file, eg a TIFF file. Is that going to be a problem for map2col? Possibly the source should be validated here to ensure it is a .map file and raise a QgsProcessingException if it's not.

Reply:
Good point. If it's not possible to limit the raster list selection in the GUI to .map files, it indeed needs to raise an exception. I've noticed that this is also the case with the following tools: _resample.py_ and _pcraster_lookup_algorithm.py_.

Reply:
So a quick and easy way to handle this would be to test if `layer.source().lower().endswith('.map')` and raise a the QgsProcessingException if this is not the case

Reply:
Okay. I've committed the change to map2col.py. Please check. Then I can also apply it to the other scripts.